packages/util/src/logger/cloud-sdk-logger.ts

```diff
@@ -181,6 +181,57 @@ export function getGlobalLogLevel(): string | undefined {
   return container.options.level;
 }
 
+const defaultSensitiveKeys = [
+  'access_token',
+  'authentication',
+  'authorization',
+  'apiKey',
+  'credentials',
+  'csrf',
+  'xsrf',
+  'secret',
+  'password',
+  'JTENANT',
+  'JSESSION'
+];
+
+/**
+ * Potentially sensitive keys will be matched case-insensitive and as substrings.
+ * Matches will be replaced with a placeholder string.
+ * @param input - The record to be sanitized.
+ * @param replacementString - The placeholder string.
+ * @param sensitiveKeys - The list of keys to be replaced. This overrides the default list.
+ * @returns The sanitized copy of the input record.
+ */
+export function sanitizeRecord<T = any>(
+  input: Record<string, T>,
+  replacementString = '<DATA NOT LOGGED TO PREVENT LEAKING SENSITIVE DATA>',
+  sensitiveKeys: string[] = defaultSensitiveKeys
+): Record<string, T> {
+  const normalizedKeys = sensitiveKeys.map(key => key.toLowerCase());
+  return Object.fromEntries(
+    Object.entries(input).map(([inputKey, value]) => {
+      if (
+        normalizedKeys.some(normalizedKey =>
```

Comment:
isn't there a more simple way to check `inputKey` is in the array `normalizedKeys`. They are all string so could you use something like: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Reply:
includes would only check if the exact same string is included. This implementation checks if any recordKey has a substring that matches the sensistiveKeys

Reply:
ah ok.